dockerfiles/theia/Dockerfile

```diff
@@ -64,12 +64,11 @@ ARG CDN_PREFIX=""
 ARG MONACO_CDN_PREFIX=""
 WORKDIR ${HOME}/theia-source-code
 
-COPY che-theia/che-theia-init-sources.yml ${HOME}/che-theia-init-sources.yml
+#COPY che-theia/che-theia-init-sources.yml ${HOME}/che-theia-init-sources.yml
+COPY che-theia/che-theia.tar.gz ${HOME}/che-theia.tar.gz
```

Comment:
why not adding the directory instead of zipping/unzipping ?

Reply:
We cannot copy files outside of docker context:
```
The <src> path must be inside the context of the build; you cannot COPY ../something /something, because the first step of a docker build is to send the context directory (and subdirectories) to the docker daemon.
```

Reply:
I think it was working with symlinks

Reply:
No, it's not, see https://github.com/moby/moby/issues/1676

Reply:
BTW it doesn't seem to work as expected

```
$ cd dockerfiles/theia
$ git ls-files
.dockerignore
.gitignore
Dockerfile
README.md
build.sh
e2e/.dockerignore
e2e/.gitignore
e2e/Dockerfile
e2e/build.sh
e2e/cypress/integration/theia/commands.spec.ts
e2e/cypress/integration/theia/typescript.spec.ts
e2e/cypress/plugins/index.js
e2e/cypress/support/commands.ts
e2e/cypress/support/index.ts
e2e/src/cypress.json
e2e/src/docker-run.sh
e2e/src/package.json
e2e/src/snapshots.js
e2e/src/tsconfig.json
e2e/src/tslint.json
e2e/src/webpack.config.js
e2e/test.sh
extract-for-cdn.sh
push-cdn-files-to-akamai.sh
src/entrypoint.sh
src/patches/README.md
```

Reply:
Have you run build from script?
On my laptop I have :
```
Compresing 'che-theia' --> /Users/yevhen/work/redhat/che-theia/dockerfiles/theia/che-theia/che-theia.tar.gz
.github/CODEOWNERS
.github/ISSUE_TEMPLATE/where-to-report-issues.md
.github/PULL_REQUEST_TEMPLATE.md
.gitignore
CONTRIBUTING.md
LICENSE
README.md
build.sh
build_and_push_docker_images.sh
che-theia-init-sources.yml
che-theia-screenshot.png
configs/base.tsconfig.json
configs/tsfmt.json
configs/tslint.json
devfiles/che-theia-all.devfile.yaml
docker_image_build.include
dockerfiles/build.include
dockerfiles/build.sh
dockerfiles/remote-plugin-dependency-analytics-0.0.12/Dockerfile
dockerfiles/remote-plugin-dependency-analytics-0.0.12/build.sh
dockerfiles/remote-plugin-dotnet-2.2.105/Dockerfile
dockerfiles/remote-plugin-dotnet-2.2.105/build.sh
dockerfiles/remote-plugin-go-1.10.7/.gitignore
dockerfiles/remote-plugin-go-1.10.7/Dockerfile
dockerfiles/remote-plugin-go-1.10.7/build.sh
dockerfiles/remote-plugin-java11/Dockerfile
dockerfiles/remote-plugin-java11/build.sh
dockerfiles/remote-plugin-java8/Dockerfile
dockerfiles/remote-plugin-java8/build.sh
dockerfiles/remote-plugin-kubernetes-tooling-0.1.17/Dockerfile
dockerfiles/remote-plugin-kubernetes-tooling-0.1.17/build.sh
dockerfiles/remote-plugin-kubernetes-tooling-1.0.0/Dockerfile
dockerfiles/remote-plugin-kubernetes-tooling-1.0.0/build.sh
dockerfiles/remote-plugin-kubernetes-tooling-1.0.0/etc/docker.sh
dockerfiles/remote-plugin-openshift-connector-0.0.17/Dockerfile
dockerfiles/remote-plugin-openshift-connector-0.0.17/build.sh
dockerfiles/remote-plugin-openshift-connector-0.0.21/Dockerfile
dockerfiles/remote-plugin-openshift-connector-0.0.21/build.sh
dockerfiles/remote-plugin-php7/Dockerfile
dockerfiles/remote-plugin-php7/build.sh
dockerfiles/remote-plugin-python-3.7.3/Dockerfile
dockerfiles/remote-plugin-python-3.7.3/build.sh
dockerfiles/theia-dev/.gitignore
dockerfiles/theia-dev/Dockerfile
dockerfiles/theia-dev/build.sh
dockerfiles/theia-dev/e2e/Dockerfile
dockerfiles/theia-dev/e2e/build.sh
dockerfiles/theia-dev/src/entrypoint.sh
dockerfiles/theia-endpoint-runtime/.gitignore
dockerfiles/theia-endpoint-runtime/Dockerfile
dockerfiles/theia-endpoint-runtime/build.sh
dockerfiles/theia-endpoint-runtime/etc/entrypoint.sh
dockerfiles/theia-endpoint-runtime/etc/package.json
dockerfiles/theia/.dockerignore
dockerfiles/theia/.gitignore
dockerfiles/theia/Dockerfile
dockerfiles/theia/README.md
dockerfiles/theia/build.sh
dockerfiles/theia/e2e/.dockerignore
dockerfiles/theia/e2e/.gitignore
dockerfiles/theia/e2e/Dockerfile
dockerfiles/theia/e2e/build.sh
dockerfiles/theia/e2e/cypress/integration/theia/commands.spec.ts
dockerfiles/theia/e2e/cypress/integration/theia/typescript.spec.ts
dockerfiles/theia/e2e/cypress/plugins/index.js
dockerfiles/theia/e2e/cypress/support/commands.ts
dockerfiles/theia/e2e/cypress/support/index.ts
dockerfiles/theia/e2e/src/cypress.json
dockerfiles/theia/e2e/src/docker-run.sh
dockerfiles/theia/e2e/src/package.json
dockerfiles/theia/e2e/src/snapshots.js
dockerfiles/theia/e2e/src/tsconfig.json
dockerfiles/theia/e2e/src/tslint.json
dockerfiles/theia/e2e/src/webpack.config.js
dockerfiles/theia/e2e/test.sh
dockerfiles/theia/extract-for-cdn.sh
dockerfiles/theia/push-cdn-files-to-akamai.sh
dockerfiles/theia/src/entrypoint.sh
dockerfiles/theia/src/patches/README.md
extensions/che-theia-hosted-plugin-manager-extension/.gitignore
extensions/che-theia-hosted-plugin-manager-extension/README.md
extensions/che-theia-hosted-plugin-manager-extension/package.json
extensions/che-theia-hosted-plugin-manager-extension/src/node/che-workspace-hosted-plugin-uri-post-processor.ts
extensions/che-theia-hosted-plugin-manager-extension/src/node/hosted-plugin-manager-contribution-extension-backend-module.ts
extensions/che-theia-hosted-plugin-manager-extension/tsconfig.json
extensions/eclipse-che-theia-about/.gitignore
extensions/eclipse-che-theia-about/package.json
extensions/eclipse-che-theia-about/scripts/generate-about-details.js
extensions/eclipse-che-theia-about/src/browser/about-che-theia-dialog.ts
extensions/eclipse-che-theia-about/src/browser/about-frontend-module.ts
extensions/eclipse-che-theia-about/src/browser/style/che-logo-dark.svg
extensions/eclipse-che-theia-about/src/browser/style/che-logo-light.svg
extensions/eclipse-che-theia-about/tsconfig.json
extensions/eclipse-che-theia-activity-tracker/.gitignore
extensions/eclipse-che-theia-activity-tracker/README.md
extensions/eclipse-che-theia-activity-tracker/package.json
extensions/eclipse-che-theia-activity-tracker/src/browser/che-theia-activity-tracker-contribution.ts
extensions/eclipse-che-theia-activity-tracker/src/browser/che-theia-activity-tracker-frontend-module.ts
extensions/eclipse-che-theia-activity-tracker/src/common/activity-tracker-protocol.ts
extensions/eclipse-che-theia-activity-tracker/src/node/activity-tracker-service.ts
extensions/eclipse-che-theia-activity-tracker/src/node/che-theia-activity-tracker-server-module.ts
extensions/eclipse-che-theia-activity-tracker/tsconfig.json
extensions/eclipse-che-theia-dashboard/.gitignore
extensions/eclipse-che-theia-dashboard/package.json
extensions/eclipse-che-theia-dashboard/src/browser/che-theia-dashboard-frontend-contribution.ts
extensions/eclipse-che-theia-dashboard/src/browser/che-theia-dashboard-module.ts
extensions/eclipse-che-theia-dashboard/src/browser/style/che-theia-dashboard-module.css
extensions/eclipse-che-theia-dashboard/src/browser/theia-dashboard-client.ts
extensions/eclipse-che-theia-dashboard/tsconfig.json
extensions/eclipse-che-theia-plugin-ext/.gitignore
extensions/eclipse-che-theia-plugin-ext/README.md
extensions/eclipse-che-theia-plugin-ext/package.json
extensions/eclipse-che-theia-plugin-ext/src/browser/che-api-provider.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/che-devfile-main.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/che-factory-main.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/che-frontend-module.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/che-ssh-main.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/che-task-client.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/che-task-main.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/che-user-main.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/che-variables-main.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/che-workspace-main.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-command-contribution.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-frontend-service.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-manager.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-menu.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-preferences.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-view-contribution.ts
extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-widget.tsx
extensions/eclipse-che-theia-plugin-ext/src/browser/style/che-plugins.css
extensions/eclipse-che-theia-plugin-ext/src/common/che-protocol.ts
extensions/eclipse-che-theia-plugin-ext/src/common/plugin/plugin-filter.ts
extensions/eclipse-che-theia-plugin-ext/src/node/che-api-service.ts
extensions/eclipse-che-theia-plugin-ext/src/node/che-backend-module.ts
extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-api-provider.ts
extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-script-service.ts
extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts
extensions/eclipse-che-theia-plugin-ext/src/node/che-task-service.ts
extensions/eclipse-che-theia-plugin-ext/src/plugin/che-api.ts
extensions/eclipse-che-theia-plugin-ext/src/plugin/che-devfile.ts
extensions/eclipse-che-theia-plugin-ext/src/plugin/che-factory.ts
extensions/eclipse-che-theia-plugin-ext/src/plugin/che-ssh.ts
extensions/eclipse-che-theia-plugin-ext/src/plugin/che-task-impl.ts
extensions/eclipse-che-theia-plugin-ext/src/plugin/che-user.ts
extensions/eclipse-che-theia-plugin-ext/src/plugin/che-variables.ts
extensions/eclipse-che-theia-plugin-ext/src/plugin/che-workspace.ts
extensions/eclipse-che-theia-plugin-ext/src/plugin/node/che-api-node-provider.ts
extensions/eclipse-che-theia-plugin-ext/src/plugin/webworker/che-api-worker-provider.ts
extensions/eclipse-che-theia-plugin-ext/tsconfig.json
extensions/eclipse-che-theia-plugin-ext/webpack.config.js
extensions/eclipse-che-theia-plugin-remote/.gitignore
extensions/eclipse-che-theia-plugin-remote/package.json
extensions/eclipse-che-theia-plugin-remote/src/node/dummy-trace-logger.ts
extensions/eclipse-che-theia-plugin-remote/src/node/hosted-plugin-remote.ts
extensions/eclipse-che-theia-plugin-remote/src/node/plugin-discovery.ts
extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote-backend-module.ts
extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote-init.ts
extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote-mapping.ts
extensions/eclipse-che-theia-plugin-remote/src/node/plugin-remote.ts
extensions/eclipse-che-theia-plugin-remote/src/node/remote-metadata-processor.ts
extensions/eclipse-che-theia-plugin-remote/src/node/server-plugin-proxy-runner.ts
extensions/eclipse-che-theia-plugin-remote/src/node/terminal-container-aware.ts
extensions/eclipse-che-theia-plugin-remote/src/node/websocket.ts
extensions/eclipse-che-theia-plugin-remote/tsconfig.json
extensions/eclipse-che-theia-plugin/README.md
extensions/eclipse-che-theia-plugin/package.json
extensions/eclipse-che-theia-plugin/src/che-proposed.d.ts
extensions/eclipse-che-theia-plugin/src/che.d.ts
extensions/eclipse-che-theia-preferences-provider-extension/.gitignore
extensions/eclipse-che-theia-preferences-provider-extension/package.json
extensions/eclipse-che-theia-preferences-provider-extension/src/browser/prefs-provider-extension-frontend-module.ts
extensions/eclipse-che-theia-preferences-provider-extension/src/browser/prefs-provider.ts
extensions/eclipse-che-theia-preferences-provider-extension/tests/mock.js
extensions/eclipse-che-theia-preferences-provider-extension/tests/prefs-provider-extension.test.ts
extensions/eclipse-che-theia-preferences-provider-extension/tsconfig.json
extensions/eclipse-che-theia-terminal/.gitignore
extensions/eclipse-che-theia-terminal/package.json
extensions/eclipse-che-theia-terminal/src/browser/contribution/exec-terminal-contribution.ts
extensions/eclipse-che-theia-terminal/src/browser/contribution/keybinding-context.ts
extensions/eclipse-che-theia-terminal/src/browser/contribution/terminal-command-filter.ts
extensions/eclipse-che-theia-terminal/src/browser/contribution/terminal-quick-open.ts
extensions/eclipse-che-theia-terminal/src/browser/server-definition/remote-connection.ts
extensions/eclipse-che-theia-terminal/src/browser/server-definition/remote-terminal-protocol.ts
extensions/eclipse-che-theia-terminal/src/browser/server-definition/terminal-proxy-creator.ts
extensions/eclipse-che-theia-terminal/src/browser/terminal-frontend-module.ts
extensions/eclipse-che-theia-terminal/src/browser/terminal-widget/remote-terminal-widget.ts
extensions/eclipse-che-theia-terminal/src/browser/terminal-widget/terminal.css
extensions/eclipse-che-theia-terminal/src/common/workspace-service.ts
extensions/eclipse-che-theia-terminal/src/node/terminal-backend-module.ts
extensions/eclipse-che-theia-terminal/src/node/workspace-service-impl.ts
extensions/eclipse-che-theia-terminal/tsconfig.json
extensions/eclipse-che-theia-user-preferences/.gitignore
extensions/eclipse-che-theia-user-preferences/package.json
extensions/eclipse-che-theia-user-preferences/src/node/che-theia-preferences-contribution.ts
extensions/eclipse-che-theia-user-preferences/src/node/che-theia-preferences-synchronizer.ts
extensions/eclipse-che-theia-user-preferences/src/node/che-theia-user-preferences-backend-module.ts
extensions/eclipse-che-theia-user-preferences/tsconfig.json
generator/.editorconfig
generator/.gitignore
generator/CONTRIBUTING.md
generator/README.md
generator/package.json
generator/src/__mocks__/command.ts
generator/src/__mocks__/yarn.ts
generator/src/cdn.ts
generator/src/cdn/base.ts
generator/src/cdn/bootstrap.ts
generator/src/cdn/html-template.ts
generator/src/cdn/webpack-customizer.ts
generator/src/cdn/webpack-loader.ts
generator/src/cdn/webpack.config.ts
generator/src/clean.ts
generator/src/cli-error.ts
generator/src/command.ts
generator/src/conf/.yarnclean
generator/src/conf/cleanup-find
generator/src/foreach_yarn
generator/src/index.ts
generator/src/init-sources.ts
generator/src/init.ts
generator/src/logger.ts
generator/src/production.ts
generator/src/repository.ts
generator/src/scripts/override-vs-loader.js
generator/src/templates/assembly-package.mst
generator/src/templates/cdn/custom-html.html
generator/src/templates/cdn/vs-loader.js
generator/src/yargs.ts
generator/src/yarn.ts
generator/tests/cdn.spec.ts
generator/tests/cdn/base.spec.ts
generator/tests/cdn/boostrap.spec.ts
generator/tests/cdn/html-template.spec.ts
generator/tests/cdn/webpack-customizer.spec.ts
generator/tests/cdn/webpack-loader.spec.ts
generator/tests/clean/clean.spec.ts
generator/tests/cli-error.spec.ts
generator/tests/command.spec.ts
generator/tests/init-sources/assembly-example/assembly-package.json
generator/tests/init-sources/assembly-example/package.json
generator/tests/init-sources/extension-empty/package.json
generator/tests/init-sources/extension-example/folder1/package.json
generator/tests/init-sources/extension-example/folder2/package.json
generator/tests/init-sources/extension-example/package.json
generator/tests/init-sources/extension-example/plugin-folder1/package.json
generator/tests/init-sources/extension-example/plugin-folder2/package.json
generator/tests/init-sources/extension-example2/package.json
generator/tests/init-sources/index.d.ts
generator/tests/init-sources/init-sources.spec.ts
generator/tests/init/init.spec.ts
generator/tests/init/root-folder/packages/core/package.json
generator/tests/logger.spec.ts
generator/tests/production/assembly/package.json
generator/tests/production/assembly/yarn.lock
generator/tests/production/production.spec.ts
generator/tests/repository.spec.ts
generator/tests/yarn/json-config-modules-folder.stdout
generator/tests/yarn/json-list-prod-excluded.stdout
generator/tests/yarn/json-list-prod-multiple-deps.stdout
generator/tests/yarn/json-list-prod-no-dep.stdout
generator/tests/yarn/json-list-prod-one-dep-children.stdout
generator/tests/yarn/json-list-prod-one-dep-duplicate.stdout
generator/tests/yarn/json-list-prod-one-dep.stdout
generator/tests/yarn/json-list-prod-webpack.stdout
generator/tests/yarn/yarn.spec.ts
generator/tsconfig.json
generator/tsfmt.json
generator/tslint.json
lerna.json
package.json
plugins/containers-plugin/.gitignore
plugins/containers-plugin/README.md
plugins/containers-plugin/package.json
plugins/containers-plugin/resources/cube.svg
plugins/containers-plugin/src/containers-plugin.ts
plugins/containers-plugin/src/containers-service.ts
plugins/containers-plugin/src/containers-tree-data-provider.ts
plugins/containers-plugin/tsconfig.json
plugins/factory-plugin/.gitignore
plugins/factory-plugin/.travis.yml
plugins/factory-plugin/README.md
plugins/factory-plugin/package.json
plugins/factory-plugin/src/devfile.ts
plugins/factory-plugin/src/ephemeral-workspace-checker.ts
plugins/factory-plugin/src/factory-initializer.ts
plugins/factory-plugin/src/factory-plugin.ts
plugins/factory-plugin/src/file-uri.ts
plugins/factory-plugin/src/git.ts
plugins/factory-plugin/src/projects.ts
plugins/factory-plugin/src/theia-commands.ts
plugins/factory-plugin/src/workspace-projects-manager.ts
plugins/factory-plugin/tests/file-uri.spec.ts
plugins/factory-plugin/tests/git.spec.ts
plugins/factory-plugin/tests/projects.spec.ts
plugins/factory-plugin/tsconfig.json
plugins/ports-plugin/.gitignore
plugins/ports-plugin/README.md
plugins/ports-plugin/__mocks__/@eclipse-che/plugin.ts
plugins/ports-plugin/package.json
plugins/ports-plugin/src/ip-converter.ts
plugins/ports-plugin/src/port-changes-detector.ts
plugins/ports-plugin/src/port-redirect-listener.ts
plugins/ports-plugin/src/port-scanner.ts
plugins/ports-plugin/src/port.ts
plugins/ports-plugin/src/ports-plugin.ts
plugins/ports-plugin/src/workspace-handler.ts
plugins/ports-plugin/src/workspace-port.ts
plugins/ports-plugin/tests/ip-converter.spec.ts
plugins/ports-plugin/tests/port-changes-detector/port-changes-detector-after.stdout
plugins/ports-plugin/tests/port-changes-detector/port-changes-detector-before.stdout
plugins/ports-plugin/tests/port-changes-detector/port-changes-detector.spec.ts
plugins/ports-plugin/tests/port-scanner/port-scanner-listen-ipv4.stdout
plugins/ports-plugin/tests/port-scanner/port-scanner-listen-ipv6.stdout
plugins/ports-plugin/tests/port-scanner/port-scanner.spec.ts
plugins/ports-plugin/tests/workspace-handler/workspace-handler.spec.ts
plugins/ports-plugin/tests/workspace-handler/workspace-output.stdout
plugins/ports-plugin/tsconfig.json
plugins/ssh-plugin/.gitignore
plugins/ssh-plugin/README.md
plugins/ssh-plugin/package.json
plugins/ssh-plugin/src/node/ssh-key-manager.ts
plugins/ssh-plugin/src/ssh-plugin-backend.ts
plugins/ssh-plugin/tsconfig.json
plugins/task-plugin/.gitignore
plugins/task-plugin/README.md
plugins/task-plugin/package.json
plugins/task-plugin/resources/preview-urls-view.css
plugins/task-plugin/resources/preview-urls.js
plugins/task-plugin/src/che-task-backend-module.ts
plugins/task-plugin/src/che-workspace-client.ts
plugins/task-plugin/src/export/export-configs-manager.ts
plugins/task-plugin/src/export/launch-configs-exporter.ts
plugins/task-plugin/src/export/task-configs-exporter.ts
plugins/task-plugin/src/machine/machine-exec-client.ts
plugins/task-plugin/src/machine/machine-exec-watcher.ts
plugins/task-plugin/src/machine/machines-picker.ts
plugins/task-plugin/src/machine/websocket.ts
plugins/task-plugin/src/preview/preview-url-open-service.ts
plugins/task-plugin/src/preview/previews-widget.ts
plugins/task-plugin/src/preview/task-events-handler.ts
plugins/task-plugin/src/preview/task-preview-mode.ts
plugins/task-plugin/src/preview/tasks-preview-manager.ts
plugins/task-plugin/src/task-plugin-backend.ts
plugins/task-plugin/src/task/che-task-provider.ts
plugins/task-plugin/src/task/che-task-runner.ts
plugins/task-plugin/src/task/task-protocol.ts
plugins/task-plugin/src/utils.ts
plugins/task-plugin/src/variable/project-path-variable-resolver.ts
plugins/task-plugin/src/variable/server-variable-resolver.ts
plugins/task-plugin/tsconfig.json
plugins/task-plugin/tsfmt.json
plugins/welcome-plugin/.gitignore
plugins/welcome-plugin/README.md
plugins/welcome-plugin/package.json
plugins/welcome-plugin/resources/che-logo.png
plugins/welcome-plugin/resources/che-logo.svg
plugins/welcome-plugin/resources/welcome-page.css
plugins/welcome-plugin/resources/welcome-page.js
plugins/welcome-plugin/src/welcome-page.ts
plugins/welcome-plugin/src/welcome-plugin.ts
plugins/welcome-plugin/tsconfig.json
set_theia_version.sh
tsconfig.json
tslint.json
yarn.lock
```
But anyway I'm going to update PR, to make sure that `git ls-files` executed from right context

Reply:
Yes I ran with build.sh of theia folder
```
$ cd dockerfiles/theia
$ ./build.sh --parameters
```